semantic-java/src/Language/Java.hs

```diff
@@ -0,0 +1,17 @@
+-- | Semantic functionality for Java programs.
+module Language.Java
+( Term(..)
+) where
+
+import qualified Language.Java.Tags as JavaTags
+import qualified Tags.Tagging.Precise as Tags
+import qualified TreeSitter.Java.AST as Java
+import qualified TreeSitter.Unmarshal as TS
+
+newtype Term a = Term { getTerm :: Java.Program a }
+
+instance TS.Unmarshal Term where
+  unmarshalNode node = Term <$> TS.unmarshalNode node
+
+instance Tags.ToTags Term where
+  tags src = Tags.runTagging src . JavaTags.tags . getTerm
```

Comment:
Can you get these from `GeneralizedNewtypeDeriving`?

Reply:
No, because we’re using a different typeclass. Can’t use `DerivingVia` either since we could parameterize the via-type by the target class, but not by the target class’ method.